lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -5495,42 +5496,43 @@ SILValue AdjointEmitter::accumulateDirect(SILValue lhs, SILValue rhs) {
   }
 }
 
-void AdjointEmitter::accumulateIndirect(
+void PullbackEmitter::accumulateIndirect(
     SILValue resultBufAccess, SILValue lhsBufAccess, SILValue rhsBufAccess) {
   // TODO: Optimize for the case when lhs == rhs.
   assert(lhsBufAccess->getType() == rhsBufAccess->getType()
-         && "Adjoints must have equal types!");
+         && "Pullbacks must have equal types!");
   assert(lhsBufAccess->getType().isAddress() &&
          rhsBufAccess->getType().isAddress()
-         && "Adjoint types must be both address types!");
+         && "Pullback types must be both address types!");
   auto loc = resultBufAccess.getLoc();
-  auto adjointTy = lhsBufAccess->getType();
-  auto adjointASTTy = adjointTy.getASTType();
+  auto pullbackTy = lhsBufAccess->getType();
```

Comment:
This one is incorrect because `adjointTy` refers to the type of an _adjoint value_ (stored in `lhsBufAccess`), which is not a function.